internal/controller/components/values.go

```diff
@@ -92,26 +92,57 @@ func getComponentValues(
 		}
 
 	case kcmv1.ProviderSveltosName:
-		componentValues = map[string]any{
-			"projectsveltos": map[string]any{
-				"registerMgmtClusterJob": map[string]any{
-					"registerMgmtCluster": map[string]any{
-						"args": []string{
-							"--labels=" + kcmv1.K0rdentManagementClusterLabelKey + "=" + kcmv1.K0rdentManagementClusterLabelValue,
-						},
-					},
+		projectsveltos := make(map[string]any)
+		projectsveltos["registerMgmtClusterJob"] = map[string]any{
+			"registerMgmtCluster": map[string]any{
+				"args": []string{
+					"--labels=" + kcmv1.K0rdentManagementClusterLabelKey + "=" + kcmv1.K0rdentManagementClusterLabelValue,
 				},
 			},
 		}
+
+		if opts.ImagePullSecretName != "" {
+			imagePatch := `- op: add
+  path: /spec/template/spec/imagePullSecrets
+  value:
+  - name: ` + opts.ImagePullSecretName
+
+			projectsveltos["classifierManager"] = map[string]any{
+				"agentPatchConfigMap": map[string]any{
+					"data": map[string]any{
+						"image-patch": imagePatch,
+					},
+				},
+			}
+		}
+
+		componentValues = map[string]any{
+			"projectsveltos": projectsveltos,
+		}
 	}
 
 	if opts.GlobalRegistry != "" {
-		globalValues := map[string]any{
-			"global": map[string]any{
-				"registry": opts.GlobalRegistry,
-			},
+		vals := make(map[string]any)
+		global := map[string]any{
+			"registry": opts.GlobalRegistry,
 		}
-		componentValues = chartutil.CoalesceTables(componentValues, globalValues)
+
+		if opts.ImagePullSecretName != "" {
+			global["imagePullSecrets"] = []map[string]any{
+				{
+					"name": opts.ImagePullSecretName,
+				},
+			}
+
+			vals["configSecret"] = map[string]any{
+				"create": false,
+				"name":   getProviderConfigSecretName(name),
+			}
+		}
+
+		vals["global"] = global
+
+		componentValues = chartutil.CoalesceTables(componentValues, vals)
```

Comment:
For all provider components (except sveltos), the values provided in the Management/Region will override the values generated in `getComponentValues` function (see lines 150-154):
```
	// for projectsveltos, we want new values to override values provided in Management spec
	if name == kcmv1.ProviderSveltosName {
		merged = chartutil.CoalesceTables(componentValues, currentValues)
	} else {
		merged = chartutil.CoalesceTables(currentValues, componentValues)
	}
```
So, if the user explicitly sets:
```
configSecret:
  name: abc
```
these values will take precedence over `vals`.

Reply:
Made it more coupled with the actual secret creation routine, this eliminates precedence issue and also will only pass these values if we actually about to create the Secret for the supported capi provider and not spam these to all providers.